plugins/aws/sts_provisioner.go

```diff
@@ -77,3 +63,260 @@ func (p STSProvisioner) Deprovision(ctx context.Context, in sdk.DeprovisionInput
 func (p STSProvisioner) Description() string {
 	return "Provision environment variables with temporary STS credentials AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, AWS_SESSION_TOKEN"
 }
+
+// getProfile returns the profile to be used on this run based on specified profile information
+func (p STSProvisioner) getProfile() string {
+	if len(p.profileName) != 0 {
+		return p.profileName
+	}
+
+	if profile := os.Getenv("AWS_PROFILE"); profile != "" {
+		return profile
+	}
```

Comment:
Shouldn't we strip this environment variable if it is being used? Is there anything that stops the AWS CLI from trying to re-assume a role using this profile?

Reply:
Ooh, that's right, very good catch!